src/devices/cpu/hcd62121/hcd62121.cpp

```diff
@@ -381,6 +626,102 @@ void hcd62121_cpu_device::state_export(const device_state_entry &entry)
 		case STATE_GENPCBASE:
 			m_rtemp = (m_cseg << 16) | m_ip;
 			break;
+		case HCD62121_R00:
+			m_debugger_temp = (m_reg[0x00] << 24) | (m_reg[0x01] << 16) | (m_reg[0x02] << 8) | m_reg[0x03];
+			break;
+		case HCD62121_R04:
+			m_debugger_temp = (m_reg[0x04] << 24) | (m_reg[0x05] << 16) | (m_reg[0x06] << 8) | m_reg[0x07];
+			break;
```

Comment:
Similarly, for this could you do:
Since you know these indices are sequential, and you know your register storage is linear, can’t you do something like:
```c++
...
#include "multibyte.h"
...
void hcd62121_cpu_device::state_export(const device_state_entry &entry)
{
	if ((entry.index() >= HCD62121_R00) && (entry.index() <= HCD62121_R7C))
	{
		m_debugger_temp = get_u32be(&m_reg[(entry.index() - HCD62121_R00) * 4]);
	}
	else switch (entry.index())
	{
...
```